recipes/custom_permissions.py

```diff
@@ -0,0 +1,14 @@
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+
+
+class IsOwner(IsAuthenticatedOrReadOnly):
+    message = 'Permission denied.'
+    SAFE_METHODS = ('GET', 'HEAD', 'OPTIONS')
+
+    def has_object_permission(self, request, view, obj):
+        return bool(
+            request.method in self.SAFE_METHODS
+            or request.user
+            and request.user.is_authenticated
+            and request.user == obj.author
+        )
```

Comment:
You can create a module named `core` and put this to that module. Because this is a permission that can be used by different modules.

permissions.py is enough for naming.

Can you please check my previous comments about IsOwner permission and find what is wrong here?